app/src/behaviors/behavior_hold_tap.c

```diff
@@ -587,9 +588,10 @@ static int position_state_changed_listener(const zmk_event_t *eh) {
     }
 
     // Store the position of pressed key for positional hold-tap purposes.
-    if ((ev->state) // i.e. key pressed (not released)
+    if (((!undecided_hold_tap->config->hold_trigger_on_release && ev->state)     // key pressed
+         || (undecided_hold_tap->config->hold_trigger_on_release && !ev->state)) // key released
```

Comment:
This can be simplified to `undecided_hold_tap->config->hold_trigger_on_release != ev->state`>

Reply:
good catch, done